### ISSUE TITLE
Documentation for DataLoader

### ISSUE THREAD

**Issue by cafenroy** [graphql-dotnet/graphql-dotnet#2319]
`DataLoader` is a great feature of your library. For Asp.Net core app, I suggest to improve the documentation with this small addition.

https://graphql-dotnet.github.io/docs/guides/dataloader/

In the setup section for dataloader. 

> 4. Add the DataLoaderDocumentListener to the DocumentExecuter.

For Asp.Net core project, this can be done instead : 

Configure the dataloader in `ConfigureServices`

```cs
public void ConfigureServices(IServiceCollection services) 
{
    ...

   services.AddDataLoader();

   ...
}
```

This information is hard to find if not documented. 

Thanks

**Comment by Shane32** [comment]
This applies only to the server project, correct @sungam3r ?  GraphQL.NET does not know about `IServiceCollection`.

**Comment by sungam3r** [comment]
Yes but these two projects are related and people often ask questions that are "on the border" of the two projects. Let's make a separate documentation page dedicated to the server project. Let it be even small and point to the server project documentation.

**Comment by cafenroy** [comment]
I did not realize that this where on another project until I found the solution. A link in the documentation would be useful.

**Comment by MrHPotter** [comment]
What is now the Proposed Solution here? Actually add this Part about the Server Interaction to the DataLoader Docs on GraphQl.NET or adding it in a seperate Server related Section (there is also a seperate Bounty for that). Or is this completely forgotten at this point?

**Comment by Shane32** [comment]
At this point the server project is now ignorant of the data loader entirely and configuration should be done through IGraphQLBuilder. The task here is documentation cleanup, appropriate to the current version. (I’m not really sure how things stand with the docs actually.)

**Comment by MrHPotter** [comment]
For my part, i could get the Dataloader Project up and running with the Dataloader Docs on the Page.
It's quite in depth and explains a lot of different Options.
I would say, as this is likely the most common use case, Promoting the DI/Class based approaches at the Top of the Docs instead of the Bottom might be a nice Idea nowadays.
I'd Imagine most people looking for a GraphQL Implementation and not a specific GraphQL Client Library, would be looking to set up a GraphQL Backend Server, especially in the Dotnet Ecosystem, which is often used for Web APIs nowadays.
I might be biased here tho, as a Dotnet Backend Webdev :D

TLDR: Some Reordering and maybe reworking in the Dataloader Docs Page, might be a good idea imo. Overall it seemed in a decent State tho.

**Comment by MrHPotter** [comment]
I would like to tackle this bounty next. Don't have the time for it next week, but from next Weekend on i think i could use, what i learned from the Sample Projects, to update it a bit and better integrate the new Sample Projects into the Docs

**Comment by Shane32** [comment]
Sounds great!

**Comment by Shane32** [comment]
@yuliuyi717-ux Your PR has been merged.  Submit an invoice to the open collective and post your open collective username here.
